pkg/backup/item_collector.go

```diff
@@ -316,34 +316,26 @@ func (r *itemCollector) getResourceItems(log logrus.FieldLogger, gv schema.Group
 			for _, s := range r.backupRequest.Spec.OrLabelSelectors {
 				orLabelSelectors = append(orLabelSelectors, metav1.FormatLabelSelector(s))
 			}
-		} else {
-			orLabelSelectors = []string{}
 		}
 
 		log.Info("Listing items")
 		unstructuredItems := make([]unstructured.Unstructured, 0)
 
 		// Listing items for orLabelSelectors
-		errListingForNS := false
 		for _, label := range orLabelSelectors {
 			unstructuredItems, err = r.listItemsForLabel(unstructuredItems, gr, label, resourceClient)
 			if err != nil {
-				errListingForNS = true
+				log.WithError(err).Error("Error listing items")
+				break
 			}
 		}
 
-		if errListingForNS {
```

Comment:
Although, after modification, there should be no unstructed object found when listing with or label selector, it's better to have an early return.